sql/information_schema/columns_table.go

```diff
@@ -152,185 +156,299 @@ func (c ColumnsTable) WithColumnDefaults(columnDefaults []sql.Expression) (sql.T
+	var rows []sql.Row
+	dbName := db.Name()
 
-		// TODO: View Definition is lacking information to properly fill out these table
-		// TODO: Should somehow get reference to table(s) view is referencing
-		// TODO: Each column that view references should also show up as unique entries as well
-		views, err := viewsInDatabase(ctx, db)
+	privSetDb := privSet.Database(dbName)
+	curPrivSetMap := getCurrentPrivSetMapForColumn(privSetDb.ToSlice(), privSetMap)
+	if dbName == sql.InformationSchemaDatabaseName {
+		curPrivSetMap["select"] = struct{}{}
+	}
+
+	err := sql.DBTableIter(ctx, db, func(t sql.Table) (cont bool, err error) {
+		rs, err := getRowsFromTable(ctx, db, t, privSetDb, curPrivSetMap, allColsWithDefaultValue)
 		if err != nil {
-			return nil, err
+			return false, err
 		}
+		rows = append(rows, rs...)
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
 
-		for _, view := range views {
-			rows = append(rows, sql.Row{
-				"def",     // table_catalog
-				db.Name(), // table_schema
-				view.Name, // table_name
-				"",        // column_name
-				uint32(0), // ordinal_position
-				nil,       // column_default
-				"",        // is_nullable
-				nil,       // data_type
-				nil,       // character_maximum_length
-				nil,       // character_octet_length
-				nil,       // numeric_precision
-				nil,       // numeric_scale
-				nil,       // datetime_precision
-				"",        // character_set_name
-				"",        // collation_name
-				"",        // column_type
-				"",        // column_key
-				"",        // extra
-				"select",  // privileges
-				"",        // column_comment
-				"",        // generation_expression
-				nil,       // srs_id
-			})
+	return rows, nil
+}
+
+// getCurrentPrivSetMapForColumn returns a new privilege set map that contains what the given privilege set map has,
+// and it adds any available privileges from given array of privilege type. For example, the given privilege set map
+// may contain general privilege types for the database only, and the given array of privilege type will contain all
+// privilege types defined for the table specifically. This function only add `select`, `insert`, `update` and
+// `references` privileges to the new privilege set map if available. These are column level privileges only.
+func getCurrentPrivSetMapForColumn(privs []sql.PrivilegeType, privSetMap map[string]struct{}) map[string]struct{} {
+	curPrivSetMap := make(map[string]struct{})
+	for p := range privSetMap {
+		curPrivSetMap[p] = struct{}{}
+	}
+	for _, pt := range privs {
+		switch pt {
+		// columns can have 'select', 'insert', 'update', 'references' privileges only.
+		case sql.PrivilegeType_Select, sql.PrivilegeType_Insert, sql.PrivilegeType_Update, sql.PrivilegeType_References:
+			priv := strings.ToLower(pt.String())
+			if _, ok := curPrivSetMap[priv]; !ok {
+				curPrivSetMap[priv] = struct{}{}
+			}
```

Comment:
(minor) I don't think you need this `if` statement... seems like you could save a line or two if you just always executed the line inside the if block. Since it's a map, it will just overwrite the current value if it's already set and will have the same outcome without the `if` check.